drivers/azure/util.go

```diff
@@ -164,6 +114,26 @@ func (d *Driver) ipAddress(ctx context.Context) (ip string, err error) {
 	return ip, nil
 }
 
+// resolveNSGReference extracts d.nsgResourceID and d.nsgUsedInPool from d.NSG
+func (d *Driver) resolveNSGReference() {
```

Comment:
maybe refactor to be in azureutil and and use like  `d.nsgResourceID, d.nsgUsedInPool= azureutil.resolveNSGReference(d.NSG)` so it's not just a parse/set on the struct so it's less black box

Reply:
Great idea, but I think it should be left in util.go. It still uses other driver values (i.e. d.SubscriptionID, d.ResourceGroup, d.naming().NSG() in the legacy case)

Reply:
sure that's fine

Reply:
oh ya makes sense uses subscription id and a bunch of things in the last return. makes sense